Classes/Lightwerk/SurfTasks/Task/Assets/BowerTask.php

```diff
@@ -0,0 +1,79 @@
+<?php
+namespace Lightwerk\SurfTasks\Task\Assets;
+
+/*                                                                        *
+ * This script belongs to the TYPO3 Flow package "TYPO3.Surf".            *
+ *                                                                        */
+
+use TYPO3\Flow\Annotations as Flow;
+use TYPO3\Surf\Domain\Model\Application;
+use TYPO3\Surf\Domain\Model\Deployment;
+use TYPO3\Surf\Domain\Model\Node;
+use TYPO3\Surf\Domain\Model\Task;
+use TYPO3\Surf\Exception\TaskExecutionException;
+
+/**
+ * @author Achim Fritz <af@lightwerk.com>
+ * @package Lightwerk\SurfTasks
+ */
+class BowerTask extends Task {
+
+	/**
+	 * @Flow\Inject
+	 * @var \TYPO3\Surf\Domain\Service\ShellCommandService
+	 */
+	protected $shell;
+
+	/**
+	 * Executes this task
+	 *
+	 * @param Node $node
+	 * @param Application $application
+	 * @param Deployment $deployment
+	 * @param array $options
+	 * @return void
+	 * @throws TaskExecutionException
+	 */
+	public function execute(Node $node, Application $application, Deployment $deployment, array $options = array()) {
+
+		if (isset($options['useApplicationWorkspace']) && $options['useApplicationWorkspace'] === TRUE) {
+			$rootPath = $deployment->getWorkspacePath($application);
+		} else {
+			$rootPath = $deployment->getApplicationReleasePath($application);
+		}
+		if (isset($options['relativeRootPath'])) {
+			$rootPath .= $options['relativeRootPath'];
+		}
+
+		if (!empty($options['nodeName'])) {
```

Comment:
This could be moved to the top since it contains an "early return" (through exception)